redis/src/main/scala/zio/redis/api/SortedSets.scala

```diff
@@ -8,18 +8,30 @@ trait SortedSets {
   final val bzPopMax = RedisCommand("BZPOPMAX", Tuple2(DurationSecondsInput, NonEmptyList(StringInput)), ChunkOutput)
   final val bzPopMin = RedisCommand("BZPOPMIN", Tuple2(DurationSecondsInput, NonEmptyList(StringInput)), ChunkOutput)
 
-  // TODO: represent two possible outputs as per the documentation
-  final val zAdd = RedisCommand(
-    "ZADD",
-    Tuple5(
-      StringInput,
-      OptionalInput(UpdateInput),
-      OptionalInput(ChangedInput),
-      OptionalInput(IncrementInput),
-      NonEmptyList(MemberScoreInput)
-    ),
-    LongOutput
-  )
+  final def zAdd =
+    RedisCommand(
+      "ZADD",
+      Tuple4(
+        StringInput,
+        OptionalInput(UpdateInput),
+        OptionalInput(ChangedInput),
+        NonEmptyList(MemberScoreInput)
+      ),
+      LongOutput
+    )
+
+  final def zAddWithIncr =
+    RedisCommand(
+      "ZADD",
+      Tuple5(
+        StringInput,
+        OptionalInput(UpdateInput),
+        OptionalInput(ChangedInput),
+        IncrementInput,
+        NonEmptyList(MemberScoreInput)
+      ),
+      StringOutput
```

Comment:
I think that this one should be `OptionalOutput(DoubleOutput)`.

Reply:
why ? if you try redis zAdd with INCR, it will return you a string type not double ;)

Reply:
Looks like it returns double as bulk string:
```
If the INCR option is specified, the return value will be Bulk string reply:

The new score of member (a double precision floating point number) represented as string, or nil if the operation was aborted (when called with either the XX or the NX option).
```

Reply:
@anovakovic01 well try in redis console ;)

btw.
https://redis.io/commands/zadd
> If the INCR option is specified, the return value will be Bulk string reply

Reply:
> 127.0.0.1:6379> ZADD set1 1 a 2 b
> (integer) 2
> 127.0.0.1:6379> ZADD set1 INCR 2 a
> "3"

Reply:
I've tried it:
```
ZADD myzset INCR 1 "one" 
$1
1
ZADD myzset INCR 4.2 "two" 
$18
4.2000000000000002
```

Also, on the same link, right below the line that you've just sent it says that it is `a double precision floating point number`, it's just represented as string.

Reply:
hmm what redis version do you have ?

Reply:
ah you are right, I see now implementation of DoubleOutput, wonder why redis-cli is not showimg me those $18 etc. will fix it in a sec

Reply:
> wonder why redis-cli is not showimg me those $18 etc. will fix it in a sec

Try sending requests using telnet, then you should get protocol level responses.